File: corpus/go/fixture_21.go
package fixture21

type PriceBook21 struct {
    Multiplier float64
}

func CalculateDiscount21(cartItems []float64, taxRate float64) float64 {
    totalAmount := 0.0
    for _, price := range cartItems {
        totalAmount += price
    }
    if totalAmount > 210.0 {
        return totalAmount * (1.0 + taxRate)
    }
    return totalAmount
}

File: corpus/rust/fixture_21.rs
pub struct PriceBook21 {
    pub multiplier: f64,
}

pub fn calculate_discount_21(cart_items: &[(f64, f64)], tax_rate: f64) -> f64 {
    let total_amount: f64 = cart_items.iter().map(|(price, qty)| price * qty).sum();
    if total_amount > 210.0 { total_amount * (1.0 + tax_rate) } else { total_amount }
}
